File: src/IotBbq.App/IotBbq.App/Services/IThermometerService.cs
using System.Threading.Tasks;
using IotBbq.App.Mcp3008;

namespace IotBbq.App.Services
{
    public interface IThermometerService
    {
        Task<Temps> ReadThermometer(int index);
    }

    public class Temps
    {
        public double Kelvin { get; set; }

        public double Celcius { get; set; }

        public double Farenheight { get; set; }
    }
}

File: src/IotBbq.App/IotBbq.JsApp/src/app/services/IThermometerService.ts
import { InjectionToken } from '@angular/core';

export interface ITemps {
  kelvin: number;
  celcius: number;
  farenheight: number;
}

export interface IThermometerService {
  readThermometer(index: number): Promise<ITemps>;
}

export const THERM_SVC_TOKEN: InjectionToken<IThermometerService> = new InjectionToken<IThermometerService>('THERM_SVC_TOKEN');
